File: result/translate/transcoder/python2java/java/ARRAY_RANGE_QUERIES_ELEMENTS_FREQUENCY_VALUE.java
public static int f_gold ( int start , int end , int [ ] arr ) {
  Map < Integer , Integer > frequency = new HashMap < Integer , Integer > ( ) ;
  for ( int i = start ;
  i <= end ;
  i ++ ) {
    if ( arr [ i ] < frequency . keySet ( ) . size ( ) ) {
      frequency . put ( arr [ i ] , 1 ) ;
    }
    else {
      frequency . put ( arr [ i ] , 1 ) ;
    }
  }
  int count = 0 ;
  for ( int x : frequency . keySet ( ) ) {
    if ( x == frequency . get ( x ) ) {
      count ++ ;
    }
  }
  return count ;
}

File: result/translate/chatgpt/java2python/python/ARRAY_RANGE_QUERIES_ELEMENTS_FREQUENCY_VALUE.py
def f_gold(start, end, arr):
    mp = {}
    for i in range(start, end+1):
        if arr[i] in mp:
            mp[arr[i]] += 1
        else:
            mp[arr[i]] = 1
    count = 0
    for key, value in mp.items():
        if key == value:
            count += 1
    return count